actblue/wordpress_plugin
# Add refcode input for embed block

This PR adds an input field to the `ActBlue Embed` block for a refcode, and sends that field along with the requested URL to the ActBlue endpoint.

## To test

0. Pull down this branch, build the assets with `yarn --cwd actblue-contributions build`, then start the environment with `docker-compose up -d`.
1. In the admin, create a new post and add an `ActBlue Embed` block and note the `Refcode` input in the sidebar:

    <img width="1011" alt="Screen Shot 2020-12-09 at 5 04 11 PM" src="https://user-images.githubusercontent.com/3286676/101707706-99590c80-3a40-11eb-8317-491a09e6a681.png">

2. In the refcode field, add some text. Then paste the following url into the block: `https://secure.actblue.com/donate/actblue-1-embed`.

    <img width="1024" alt="Screen Shot 2020-12-09 at 5 05 43 PM" src="https://user-images.githubusercontent.com/3286676/101707802-cb6a6e80-3a40-11eb-99f4-ba10d0f0d39b.png">

3. Open your browser's "Network" tab, then click the "Embed" button. The embedded form should load and you should see a request to WordPress's oEmbed endpoint; the request will send as a parameter to that endpoint the embed url you added with your refcode appended as a query parameter:

    ![Screen-Shot-2020-12-09-at-5-07-54-PM](https://user-images.githubusercontent.com/3286676/101708186-9a3e6e00-3a41-11eb-8f0c-bb22e308942f.png)

4. Save the post, then reload the page. Select the embed block and notice that the refocde is still there and unchanged.

5. Update the refcode with different text string and click the "Update" button. Notice that the embedded form preview will reload and another request will appear in the "Network" tab with the correct refcode.

## Todo

- [x] Better UI around prompting a user to click the `Update` button if they update their refcode value after an initial embed?
